cmd/fever/cmds/run.go

```diff
@@ -544,7 +545,9 @@ func mainfunc(cmd *cobra.Command, args []string) {
 		sinput.(*input.RedisInput).SubmitStats(pse)
 	} else {
 		inputSocket := viper.GetString("input.socket")
-		sinput, err = input.MakeSocketInput(inputSocket, inputChan)
+		bufDrop := viper.GetBool("input.buffer-drop")
+		sinput, err = input.MakeSocketInput(inputSocket, inputChan, bufDrop)
+		sinput.(*input.SocketInput).SubmitStats(pse)
```

Comment:
Why not var sinput *input.SocketInput in the first place?

Reply:
Because it is used later as an `input.Input` interface (by calling `sinput.SetVerbose()` and `Run()`) and I needed to call both specific functions (`SubmitStats`) as well as interface ones (`Run()`) so that's why I am "downcasting" here. Will make it a bit more explicit in an upcoming commit.